SpyGame/src/edu/cpp/cs/cs141/final_proj/UserInterface.java

```diff
@@ -184,14 +184,14 @@ else if (USER_COMMAND.abbreviatedKeyCodes().containsKey(userInput)) {
 		// User entered a direction to move in
 		if (moveDir != null) {
 			MoveStatus moveStatus = null;
-			while (true) {
-				moveStatus = game.playerMove(moveDir);
-				System.out.println("Move Status: " + moveStatus.msg);
-				if (moveStatus.moveResult != MOVE_RESULT.ILLEGAL)
-					return;
-				else
-					System.out.println(game.displayBoard());
-					moveDir = getUserDirection("move");
+
+			moveStatus = game.playerMove(moveDir);
+			System.out.println("Move Status: " + moveStatus.msg);
+			
+			// ILLEGAL move attempted, print grid & call this method again
+			if (moveStatus.moveResult == MOVE_RESULT.ILLEGAL) {
+				System.out.println(game.displayBoard());
+				playerTurn();
```

Comment:
That was pretty sneaky.  Could you put in a comment making it more obvious like //RECURSION